dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1362,118 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if self._profile[col].profile['data_type'] not in ['int', 'float'] \
+                or self._profile[col].profile['statistics']['null_count'] > 0:
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        correlation_matrix = data.corr()
+
+        if self.correlation_matrix is None:
+            self.correlation_matrix = correlation_matrix
+            return
```

Comment:
i would suggest putting this at the top and put the warning inside the `if`. That way, instead of running through  all the calculations, we skip it which would otherwise be a waste of time.

Reply:
we can leave the return in the if, but no need for it at end of function

Reply:
Also, if we skip correlations because they exist, should we set it to something besides `None`, but also not the same as the previous corr matrix? i'm not sure it is safe for them to assume the correlations are correct after an update.

Reply:
restructured the if. 
For the 'updated' correlation, I don't know what's the appropriate value to set. Currently, I keep it as the previous value, with the warning, then the users cannot use it anyway. But I'm open to a discussion on this. Plus, we'll have to deal with this shortly, so, it'd be temporally fine to leave it that way imo.

Reply:
Either way we leave it, I think the if should go above the calculations as we don't need to waste compute if we aren't even going to use the values anyways.